js/indodax.js

```diff
@@ -593,4 +519,82 @@ module.exports = class indodax extends Exchange {
         this.throwBroadlyMatchedException (this.exceptions['broad'], error, feedback);
         throw new ExchangeError (feedback); // unknown message
     }
+
+    async fetchMarkets (params = {}) {
+        const markets = await this.publicGetPairs (params);
+        // [
+        //     {
+        //         "id": "btcidr",
+        //         "symbol": "BTCIDR",
+        //         "base_currency": "idr",
+        //         "traded_currency": "btc",
+        //         "traded_currency_unit": "BTC",
+        //         "description": "BTC/IDR",
+        //         "ticker_id": "btc_idr",
+        //         "volume_precision": 0,
+        //         "price_precision": 1000,
+        //         "pricescale": 1000,
+        //         "trade_min_base_currency": 10000,
+        //         "trade_min_traded_currency": 7.321e-5,
+        //         "has_memo": false,
+        //         "memo_name": false,
+        //         "has_payment_id": false,
+        //         "trade_fee_percent": 0.3,
+        //         "url_logo": "https:/indodax.com/v2/logo/svg/color/btc.svg",
+        //         "url_logo_png": "https://indodax.com/v2/logo/png/color/btc.png",
+        //         "is_maintenance": 0
+        //     }
+        // ]
+        const result = [];
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            const id = this.safeString (market, 'id');
+            const baseId = this.safeString (market, 'traded_currency_unit');
+            const quoteId = this.safeString (market, 'base_currency');
+            let base = baseId.toUpperCase ();
+            let quote = quoteId.toUpperCase ();
+            base = this.safeCurrencyCode (base);
+            quote = this.safeCurrencyCode (quote);
+            const symbol = base + '/' + quote;
+            const maker = 0;
+            const taker = this.safeFloat (market, 'trade_fee_percent');
+            const isMaintenance = this.safeInteger (market, 'is_maintenance');
+            const active = (isMaintenance) ? 0 : 1;
+            const pricePrecision = (quoteId === 'idr') ? 0 : 8;
```

Comment:
↑ These precisions look wrong to me, because of this:

- https://github.com/ccxt/ccxt/issues/4183#issuecomment-639261530
- https://github.com/ccxt/ccxt/issues/4183#issuecomment-639272966
- https://github.com/ccxt/ccxt/issues/4183#issuecomment-642161813

Reply:
I have updated `pricePrecision` it will take its value from `price_round` fields.

[#4183 (comment)](https://github.com/ccxt/ccxt/issues/4183#issuecomment-649878390)